src/AzureResourceProvider.ts

```diff
@@ -29,6 +30,7 @@ export class AzureResourceProvider implements AppResourceProvider {
             type: nonNullProp(resource, 'type'),
             kind: resource.kind,
             location: resource.location,
+            azExtResourceType: getAzExtResourceType({ type: nonNullProp(resource, 'type'), kind: resource.kind }) ?? resource.type as AzExtResourceType,
```

Comment:
This bit is kinda concerning:

> `?? resource.type as AzExtResourceType`

I brought that up yesterday as an option but I can see it coming back to bite us.

Reply:
I agree that it could cause issues, but I'm not sure what alternatives there are. We could add an `Unknown` `AzExtResourceType`.

Reply:
Is it really so bad if it returns undefined? We could alter the API to note that it's not always defined in the `AppResource`.

Reply:
No, I think making `azExtResourceType` an optional property on `AppResource` is a good idea

Reply:
Ok, `azExtResourceType` is now optional.